django/db/models/options.py

```diff
@@ -234,7 +235,7 @@ def _prepare(self, model):
                 if already_created:
                     field = already_created[0]
                 field.primary_key = True
-                self.setup_pk(field)
+                self.setup_pk(field, promoting_parent=True)
```

Comment:
Is there a reason you can't rely on the `_meta.pk.parent_link and not _meta.pk.auto_created` instead of introducing this new flag?